ARCHITECTURE.md

```diff
@@ -0,0 +1,275 @@
+# HyperDB Architecture
+
+HyperDB is a scalable peer-to-peer key-value database.
+
+## Filesystem metaphor
+
+HyperDB is structured to be used much like a traditional hierarchical
+filesystem. A value can be written and read at locations like `/foo/bar/baz`,
+and the API supports querying or tracking values at subpaths, like how watching
+for changes on `/foo/bar` will report both changes to `/foo/bar/baz` and also
+`/foo/bar/19`.
+
+## Set of append-only logs
+
+A HyperDB is fundamentally a set of
+[hypercore](https://github.com/mafintosh/hypercore)s. A *hypercore* is a secure
+append-only log that is identified by a public key, and can only be written to
+by the holder of the corresponding private key. Because it is append-only, old
```

Comment:
what happens if I try writing to it with the wrong private key?

Reply:
If you bypassed hyperdb's and hypercore's API and wrote data to a hypercore directly using a different keypair, other peers would get very confusing when replicating with you. Right now there would probably be errors -- maybe even it'd put things in a bad / nonworking state. But what it essentially means it that other peers would immediately notice that the signing key is different than the one used for the rest of the feed and peers would probably discard the bogus entry.

Reply:
Yea, replication would fail with a "bad signature" error